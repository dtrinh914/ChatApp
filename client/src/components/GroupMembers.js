import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import {makeStyles} from '@material-ui/styles';
import uuid from 'uuid/v4';

const useStyles = makeStyles({
    list:{
        margin: 0
    },
    avatar:{
        marginRight: '10px'
    }
})

export default function GroupMembers({groupMembers}) {
    const classes = useStyles();
    return (
        <div>
            <h2>Group Members</h2>
            <List className={classes.list}>
                {groupMembers.map(member => <ListItem key={uuid()} data-testid='group-member'>
                                                <Avatar className={classes.avatar}>{member.username[0].toUpperCase()}</Avatar>
                                                <span>{member.username}</span>
                                            </ListItem>)}
            </List>
        </div>
    )
}
