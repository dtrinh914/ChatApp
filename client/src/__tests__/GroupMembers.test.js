import React from 'react';
import {cleanup, render, getAllByTestId} from '@testing-library/react';
import GroupMembers from '../components/GroupMembers';
import '@testing-library/jest-dom/extend-expect';

const seedData = [
                    {_id: "5e28c5c185ffc434b337c266", username: "test_user1"},
                    {_id: "5e28c5bc85ffc434b337c264", username: "test_user2"},
                    {_id: "5e28c5c485ffc434b337c268", username: "test_user3"},
                    {_id: "5e28c5c385ffc434b337c267", username: "test_user4"}
                 ];

afterEach(cleanup);

describe('GroupMember component with no data', () => {
    it('should match snapshot', ()=>{
        const {container} = render(<GroupMembers groupMembers={[]} />);
        expect(container).toMatchSnapshot();
    });
})

describe('GroupMember component with seedData', ()=>{
    const {container} = render(<GroupMembers groupMembers={seedData} />);
    const listItems = getAllByTestId(container, 'group-member');

    it('should match snapshot 2', () => {
        expect(container).toMatchSnapshot();
    });

    it('should return 4 list items', () => {
        expect(listItems.length).toBe(4);
    });

    it('should return list items with proper names',()=>{
        expect(listItems[0]).toHaveTextContent('test_user1');
        expect(listItems[1]).toHaveTextContent('test_user2');
        expect(listItems[2]).toHaveTextContent('test_user3');
        expect(listItems[3]).toHaveTextContent('test_user4');
    });
})


