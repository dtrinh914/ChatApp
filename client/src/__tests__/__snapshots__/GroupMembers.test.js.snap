// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupMember component with no data should match snapshot 1`] = `
<div>
  <div>
    <h2>
      Group Members
    </h2>
    <ul
      class="MuiList-root makeStyles-list-1 MuiList-padding"
    />
  </div>
</div>
`;

exports[`GroupMember component with seedData should match snapshot 2 1`] = `<div />`;
